### PR TITLE
8280841: Update SQLite to 3.37.2

### PR DESCRIPTION
**Pull request by HimaBinduMeda**
Updated SQLite to 3.37.2 released on 2022-01-06 from the current version 3.32.3.

Website : https://www.sqlite.org/index.html
Source code can be found from https://www.sqlite.org/download.html at  [sqlite-amalgamation-3370200.zip](https://www.sqlite.org/2022/sqlite-amalgamation-3370200.zip)

Verified the updated version with unit tests and sanity testing.
No issues are observed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280841](https://bugs.openjdk.java.net/browse/JDK-8280841): Update SQLite to 3.37.2

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Ambarish Rapte](https://openjdk.java.net/census#arapte) (@arapte - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/727/head:pull/727` \
`$ git checkout pull/727`

Update a local copy of the PR: \
`$ git checkout pull/727` \
`$ git pull https://git.openjdk.java.net/jfx pull/727/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 727`

View PR using the GUI difftool: \
`$ git pr show -t 727`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/727.diff">https://git.openjdk.java.net/jfx/pull/727.diff</a>

</details>

### CODE REVIEW

**Comment by arapte** [comment]
The source files match with https://www.sqlite.org/2022/sqlite-amalgamation-3370200.zip
Sanity testing looks good.
I would recommend to add UPDATING.txt to document the steps. You can refer: [libxml UPDATING.txt](https://github.com/openjdk/jfx/blob/f326e78ffdfcbbc9085bc50a38e0b4454b757157/modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt) , [icu UPDATING.txt](https://github.com/openjdk/jfx/blob/f326e78ffdfcbbc9085bc50a38e0b4454b757157/modules/javafx.web/src/main/native/Source/ThirdParty/icu/UPDATING.txt)

**Review by kevinrushforth — COMMENTED** [review]
Sanity testing looks good. I left one comment on `UPDATING.txt`.

File: modules/javafx.web/src/main/native/Source/ThirdParty/sqlite/UPDATING.txt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    - Retain JavaFX changes, if any.
+
+5. Update the new version in VERSION.txt.
+
+6. Perform build and sanity testing.
+
+7. Helper commands for removing tabs and trailing whitespaces from source files(.h and .c).
+   7.1 > cd modules/javafx.web/src/main/native/Source/ThirdParty/sqlite
+   7.2 Remove tabs from source files:
+     > sudo apt install moreutils
```
[the remark this answers is not in the document]
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/sqlite/UPDATING.txt]: Updated as per the comments

**Review by kevinrushforth — APPROVED** [review]

**Review by arapte — APPROVED** [review]

**Pull request closed without merging**